### PR TITLE
PLT-150 Add finance core api

### PR DESCRIPTION
**Pull request by barkerxuan**
This PR adds finance core api

### CODE REVIEW

File: src/FinanceCoreApi.php [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+
+use BrighteCapital\Api\Models\ProductConfig;
+use Fig\Http\Message\StatusCodeInterface;
+
+class FinanceCoreApi extends \BrighteCapital\Api\AbstractApi
+{
+
+    public const PATH = '../v2/finance';
+
+    public function getProductConfig(string $slug, string $vendorId = null, int $version = null): ?ProductConfig
```
Comment (microclimate) [inline src/FinanceCoreApi.php:15]: I would probably call this 'getFinancialProducts' but I recognize you are matching the resolver names so its fine.

**Review by microclimate — DISMISSED** [review]

**Comment by microclimate** [comment]
This looks good to me Barker, but I'm not sure about why TravisCI is failing.  I'm kind of fumbling around github's UI after so long in BB.  Is there any output from those pipe steps?

**Comment by barkerxuan** [comment]
@microclimate [Travis](https://github.com/brighte-capital/brighte-api-sdk-php/pull/26/checks?check_run_id=4834500334). Have on clue why the style check fails on PHP version 7.1.

**Comment by darkbluesun** [comment]
There are a large number of warnings about use of HEREDOC to initialize class properties with variables in them. This combined with multi-version support might be asking too much.
https://www.php.net/manual/en/language.types.string.php#language.types.string.syntax.heredoc

File: src/FinanceCoreApi.php [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+                    maxRepaymentYear
+                    forceCcaProcess
+                    defaultPaymentCycle
+                    invoiceRequired
+                    manualSettlementRequired
+                    fpBranch
+                    fpAccountType
+                }
+            }
+        GQL;
```
Comment (darkbluesun) [inline src/FinanceCoreApi.php:44]: I think this has to be at the begining of the line, not indented in php7.1

File: src/FinanceCoreApi.php (hunk 1 above)
Comment (darkbluesun) [reply src/FinanceCoreApi.php:14]: why would you call it that?

**Review by darkbluesun — APPROVED** [review]
Looks great, Barker!

**Pull request merged**